Update PO Token Guide (#71)
yt-dlp/yt-dlp-wiki: PO Token Guide.md

# YouTube PO Token Guide

> [!TIP]

> > TL;DR recommended setup: Use a [PO Token Provider plugin](#po-token-provider-plugins) to provide the `mweb` client with a [PO Token for GVS requests](#cases-where-a-po-token-is-required).

Reviewer: yeah thanks just wanted coletdjnz to know about it

YouTube is at present rolling out changes to enforce PO Tokens for video playback. Currently, only GVS and Subs require PO Tokens for some clients.

| Client         | PO Token Required For | Notes                                                                                                             |
|----------------|-----------------------|-------------------------------------------------------------------------------------------------------------------|
| `web`          | Subs, GVS             | Only SABR formats available                                                                                       |
| `web_safari`   | GVS*                  | Same as `web`. *Provides HLS (m3u8) formats which do not require PO Token for GVS at this time.                   |
| `mweb`         | GVS                   |                                                                                                                   |

> | `tv`           | Not required          | All formats may have DRM if you request too much. Requires an account. Only SABR formats available in some cases. |

Reviewer 2: I think we'll have it until we have a native PO token provider or until we have SABR support, at which time I imagine this wiki will need to be updated anyways. Their notes differ, too:
```suggestion
| `tv`           | Not required          | All formats usually have DRM unless logged-in. Only SABR formats available in some cases.                         |
| `tv_downgraded`| Not required          | Requires an account.                                                                                              |
```

| `web_safari`   | Subs, GVS*            | Same as `web`. *Provides HLS (m3u8) formats which do not require PO Token for GVS at this time.                   |
| `mweb`         | GVS                   |                                                                                                                   |
| `tv`           | Not required          | All formats may have DRM if you request too much. Requires an account. Only SABR formats available in some cases. |
| `tv_simply`    | GVS                   | Account cookies not supported                                                                                     |
| `tv_embedded`  | Not required          | Requires account cookies                                                                                          |
| `web_embedded` | Not required          | Only embeddable videos available                                                                                  |
| `web_music`    | GVS                   |                                                                                                                   |

> | `web_creator`  | GVS                   | Requires account cookies                                                                                          |

Reviewer 3: From my tests, it indeed needs a po token for formats but not for subs.

> | `web_creator`  | GVS                   | Requires account cookies                                                                                          |

Reviewer 2: Not required for subs

A PO Token is generated by either BotGuard (Web), DroidGuard (Android), iOSGuard (iOS). A PO Token from one platform cannot be used on another (i.e., Web PO Token cannot be used on Android or iOS).

> PO Tokens are bound to a content binding, such as the user session (Visitor ID or account Session ID) or the video ID. These days most PO Tokens (such as for web GVS/Player) are bound to the video ID, so a new token is required for each video.

Reviewer 2: ```suggestion
PO Tokens have a "content binding", meaning they are bound to the user session (Visitor ID or account Session ID) or to the video ID. Most PO Tokens (such as for `web` GVS/Player) are bound to the video ID, so a new token is required for each video.
```

> These PO Tokens are only valid for a limited time (at least 12 hours), so it will need to be refreshed periodically. However, some reports suggest that the token may be valid for several months.

Reviewer 2: ```suggestion
These PO Tokens have a limited lifespan (possibly as short as 12 hours), so they will need to be refreshed periodically. However, some reports suggest that the token may be valid for several months.
```
